### DISCUSSION TITLE
File Picker / How do load file from current dir?

### DISCUSSION THREAD

**Discussion by Robert-M-Muench** [helix-editor/helix#1463] [Q&A]
When I start HX from any location and call the file picker, it only shows files from subdirectories but not the working directory.

Typing in the filename of a file from the working directory, gives an empty list. Using the `:open` command, I can see the files from the working directory. That's inconsistent.

Is this is a bug or an intended behavior?

**Comment by Robert-M-Muench** [comment]
The problem seems that the file picker shows 8192 files at maximum. My folder has 10399 files.

The huge number of files is because of `nodes_modules` How to best ignore these files? Or even better, is there an option to toggle if the `nodes_modules` folder is searched or not?

**Reply by kirawi** [reply]
`.gitignore` could work (there needs to be a `.git` folder as well, but it can be empty).

**Reply by kirawi** [reply]
Though maybe we should do something about the 8192 file limit as well.

**Reply by Robert-M-Muench** [reply]
Well, that's a pretty heavy hack. Any plans/issues to improve file picker's behavior?

It would be cool to use something like https://github.com/Canop/broot (done in Rust too).

**Reply by kirawi** [reply]
It's currently planned as a plugin https://github.com/helix-editor/helix/issues/200
But I was thinking of also having VSCode-like behaviour in which one can ignore files from the file picker: https://github.com/helix-editor/helix/issues/280#issuecomment-942637706

**Reply by kirawi** [reply]
Seems like if there's a `.git` folder present, the file picker doesn't have a 8192 file limit.

**Reply by Robert-M-Muench** [reply]
Ok, going to try that.

**Reply by dylrich** [reply]
If you don't want to use `.gitignore`, you can use a plain `.ignore` file in your working directory as well.
